include/bluetooth/conn.h

```diff
@@ -328,6 +328,15 @@ struct bt_conn_le_create_param {
 	 *  Set zero to use same as LE 1M PHY scan window.
 	 */
 	u16_t window_coded;
+
+	/** Connection initiation timeout
+	 *
+	 *  Set zero to use the default timeout:
+	 *      K_SECONDS(:option:`CONFIG_BT_CREATE_CONN_TIMEOUT`)
+	 *
+	 *  @note Unused in @ref bt_conn_create_auto_le
+	 */
+	s32_t timeout;
```

Comment:
We should consider changing this to `k_timeout_t`

Reply:
The timeout value for the advertiser is u16_t, that is more because we use HCI units since it is an HCI parameter, but we could match it here.  `u16_t timeout;  /* N * 0.625 ms */`

Reply:
I'd be fine with that.

Reply:
Looking at this now, I can't see any mechanism for comparing two timeouts except for equality.
Additionally some of the timeout functions are still returning s32_t, like k_delayed_work_remaining_get

Reply:
I think lack of comparison helpers is an oversight, I think. I.e. would be good if someone added those so that they're available for all Zephyr users.

Reply:
@joerchan  That would limit the maximum timeout to ~40 seconds, well below the 1-65535 range specified by the Kconfig symbol

Reply:
@JordanYates Sorry, I picked the wrong unit. I meant to use the same timeout unit as the advertiser timeout. `u16_t timeout /** Advertiser timeout (N * 10 ms).  */`, that should give us sufficient range.
I think the upper limit of 65535 is arbitrary, we probably can skip the upper limit and only check against RPA_TIMEOUT when privacy is enabled.

Reply:
I'll check if it is possible to add comparison helpers, but it may have been intentional by them to only have equality comparison, the comment on that helper says it should be used to compare against K_FOREVER or K_NO_WAIT.